front/src/applications/operationalStudies/utils.ts

```diff
@@ -113,64 +113,73 @@ export const transformBoundariesDataToRangesData = <
   return formattedData;
 };
 
-export const formatElectrificationRanges = (
+const rangeContainsSegment = (
+  range: { start: number; stop: number },
+  segmentStart: number,
+  segmentEnd: number
+) => range && (range.start <= segmentStart || isNaN(range.start)) && range.stop >= segmentEnd;
+
+export const mergeElectrificationAndProfiles = (
   electrifications: ElectricalRangesData<ElectrificationValue>[],
   electricalProfiles: ElectricalRangesData<ElectricalProfileValue>[]
-): ElectrificationRange[] =>
-  // Electrifications can be of three types, electricalProfiles only two, so we know electrifications
-  // will always be at least as long as electricalProfiles so we can use it as the main array
-  electrifications.map((currentElectrification, index) => {
-    const currentProfile = electricalProfiles[index];
-
-    // currentProfile is defined as long as we didn't reach the end of electricalProfiles array
-    if (currentProfile) {
-      // If start and stop are identical, we can merge the two items
-      if (
-        currentElectrification.start === currentProfile.start &&
-        currentElectrification.stop === currentProfile.stop
-      ) {
-        return {
-          electrificationUsage: {
-            ...currentElectrification.values,
-            ...currentProfile.values,
-          },
-          start: currentElectrification.start,
-          stop: currentElectrification.stop,
-        };
-      }
-
-      // Find the profile matching the current electrification range
-      // We know we will find one because currentProfile is still defined
-      const associatedProfile = electricalProfiles.find(
-        (profile) => profile.stop >= currentElectrification.stop
-      )!;
-
-      return {
-        electrificationUsage: {
-          ...currentElectrification.values,
-          ...associatedProfile.values,
-        },
-        start: currentElectrification.start,
-        stop: currentElectrification.stop,
-      };
+): ElectrificationRange[] => {
+  const sortedElectrifications = [...electrifications];
+  sortedElectrifications.sort((e1, e2) => e1.start - e2.start);
+  const sortedProfiles = [...electricalProfiles];
+  sortedProfiles.sort((e1, e2) => e1.start - e2.start);
+
+  const boundaries = Array.from(
+    new Set(
+      sortedElectrifications
+        .flatMap((range) => [range.start, range.stop])
+        .concat(sortedProfiles.flatMap((range) => [range.start, range.stop]))
+    )
+  ).sort((a, b) => a - b);
```

Comment:
```suggestion
   const boundaries = sortedElectrifications
    .flatMap((range) => [range.start, range.stop])
    .concat(sortedProfiles.flatMap((range) => [range.start, range.stop]))
    .sort((a, b) => a - b);
```


not sure about that, but do you really need to set?

Reply:
Good catch! Since we handle 0 zero length segments anyway, it should be fine to have some redundant boundaries here.